src/parse_quote.rs

```diff
@@ -0,0 +1,156 @@
+/// Quasi-quotation macro that accepts input like the [`quote!`] macro but uses
+/// type inference to figure out a return type for those tokens.
+///
+/// [`quote!`]: https://docs.rs/quote/0.4/quote/index.html
+///
+/// The return type can be any syntax tree node that implements the [`Synom`]
```

Comment:
Perhaps mention the special cases here under a subheading?

Reply:
Yep I have a "Special cases" heading below. I would prefer to keep them below the examples because most people are not going to keep straight which types do and don't implement Synom. They will just see the examples, understand the concept, and then try it with Attribute and have it silently do the thing they wanted.